src/runtime_src/core/edge/user/aie_sys.h

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef _XCL_AIE_SYS_H_
+#define _XCL_AIE_SYS_H_
+
+#include <string>
+#include <vector>
+#include <fstream>
+
+namespace aie_sys_parser {
+
+static std::fstream sysfs_open_path(const std::string& path, std::string& err,
```

Comment:
The function definitions in this **header** file should not be static, nor should the be global symbols.   If this header file is included from multiple compilation units, I think you will get multiple defined symbols.   If these functions are meant to be defined in the header as is the case here, then they should all be marked inline.

Reply:
For me personally, I rarely place the body of a function in a header file for I do believe it is a bad practice.  I do know that there are exceptions and when they do occur, a comment is added to the function indicating why it was necessary to do so.

Reply:
fixed